<launch>
    <env name="ROSCONSOLE_FORMAT" value="[${severity}] [${walltime}] [${node}]: ${message}" />

    <arg name="gui" default="true"/>
    <arg name="noplugin" default="false"/>
    <arg name="physics_engine" default="bullet"/>
    <arg name="gazebo_seed" default="20200413"/>
    <arg name="camera_width" default="213"/>
    <arg name="camera_height" default="120"/>
    <arg name="wall_sim_speed" default="false"/>
        
    <include file="$(find lr_gym)/launch/gazebo_server.launch">
        <arg name="gui" default="$(arg gui)"/>
        <arg name="paused" default="true"/>
        <arg name="physics_engine" default="$(arg physics_engine)"/>
        <arg name="limit_sim_speed" default="false"/>
        <arg name="world_name" value="$(find lr_gym)/worlds/ground_plane_world_plugin.world"/>
        <arg name="gazebo_seed" value="$(arg gazebo_seed)"/>
        <arg name="wall_sim_speed" value="$(arg wall_sim_speed)"/>
    </include>

      <!-- publish the model URDF on the parameter server -->
    <param name="cartpole_v0_robot_description"
           command="$(find xacro)/xacro $(find lr_gym)/models/cartpole_v0.urdf.xacro
                    camera_width:=$(arg camera_width)
                    camera_height:=$(arg camera_height)" />

    <!-- spawn in gazebo using the model description in the parameter server (when gazebo starts) -->
    <node name="spawn_model" pkg="gazebo_ros" type="spawn_model" respawn="false"
          output="screen" args="-param cartpole_v0_robot_description -urdf -x 0 -y 0 -z 0.001 -model cartpole_v0" />



</launch>